### ISSUE TITLE
CA1822 Code fix strips the async keyword.

### ISSUE THREAD

**Issue by brian-reichle** [dotnet/roslyn-analyzers#2888]
#### Analyzer package

FxCopAnalyzers

#### Package Version

2.9.4 & 2.9.5

#### Diagnostic ID

[CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822-mark-members-as-static?view=vs-2019)

#### Repro steps

Enter the following code, wait for the rule to identify that the BarAsync method could be static and then apply the code fix.

```csharp
public async Task<int> BarAsync()
{
    await Task.Delay(20).ConfigureAwait(false);
    return 20;
}
```

#### Expected behavior

I would expect it to simply add the static keyword,

```csharp
public static async Task<int> BarAsync()
{
    await Task.Delay(20).ConfigureAwait(false);
    return 20;
}
```

#### Actual behavior

The current behaviour is that static keyword replaces the async keyword, resulting in compiler errors.

```csharp
public static Task<int> BarAsync()
{
    await Task.Delay(20).ConfigureAwait(false);
    return 20;
}
```

**Comment by AartBluestoke** [comment]
just ran into this one also. you don't need a generic to trigger this
`    public class TestCase
    {
        public async Task f()
        { }
    }`
![image](https://user-images.githubusercontent.com/1892829/66440205-fa243800-ea7d-11e9-826b-0ea412953e51.png)
